server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -414,6 +440,37 @@ export class MongoCollection<T> implements core.ICollection<T>, core.IRetryable
 			}
 		}
 	}
+
+	private terminateBasedOnCounterThreshold(counters: Record<string, number>): void {
+		if (this.apiFailureRateTerminationThreshold > 1) {
+			return; // If threshold set more than 1, meaning we should ever terminate and skip followings.
```

Comment:
nit: did you mean "If threshold set more than 1, meaning we should **never** [...]"?